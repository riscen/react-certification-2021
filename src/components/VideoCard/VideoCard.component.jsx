import React, { useRef } from 'react';
import { Link } from 'react-router-dom';
import { Paper } from '@material-ui/core';

import {
  VideoThumbnail,
  VideoInfo,
  VideoTitle,
  VideoDescription,
} from './VideoCard.module';

import './VideoCard.styles.css';

function HomeView(props) {
  const { title, thumbnail, description, videoId } = props;
  const cardRef = useRef(null);

  return (
    <Link to={`/video/${videoId}`}>
      <Paper className="video-card" elevation={3} ref={cardRef}>
        <VideoThumbnail>
          <img alt={title} src={thumbnail} />
        </VideoThumbnail>
        <VideoInfo>
          <VideoTitle>
            <span>{title}</span>
          </VideoTitle>
          <VideoDescription>{description || 'No description :('}</VideoDescription>
        </VideoInfo>
      </Paper>
    </Link>
  );
}

export default HomeView;
