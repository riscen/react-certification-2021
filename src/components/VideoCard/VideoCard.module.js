import styled from 'styled-components';

export const VideoThumbnail = styled.div`
  height: auto;
  width: 100%;
`;

export const VideoInfo = styled.div`
  padding: 0 20px;
  text-align: left;
`;

export const VideoTitle = styled.div`
  font-size: 18px;
  font-weight: bold;
  display: -webkit-box;
  text-overflow: ellipsis;
  overflow: hidden;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
`;

export const VideoDescription = styled.div`
  font-size: 12px;
  font-weight: normal;

  &:hover {
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
  }
`;
