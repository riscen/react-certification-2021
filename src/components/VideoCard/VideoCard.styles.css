.video-card {
  background-color: #fff;
  border-radius: 4px;
  -webkit-box-shadow: 0px 10px 13px -7px #000000, 1px 1px 8px 1px rgba(0, 0, 0, 0);
  box-shadow: 0px 10px 13px -7px #000000, 1px 1px 8px 1px rgba(0, 0, 0, 0);
  cursor: pointer;
  height: 290px;
  max-height: 290px;
  margin: 20px;
  overflow: hidden;
  padding-bottom: 10px;
  width: 300px;
  transition: max-height 0.2s ease-out;
  -webkit-transition: max-height 0.2s ease-out;
}

.video-card:hover {
  height: 350px;
  max-height: 350px;
  transition: max-height 0.2s ease-in;
  -webkit-transition: max-height 0.2s ease-in;
}
