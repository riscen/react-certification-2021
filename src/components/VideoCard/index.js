export { default } from './VideoCard.component';
export {
  VideoThumbnail,
  VideoInfo,
  VideoTitle,
  VideoDescription,
} from './VideoCard.module';
