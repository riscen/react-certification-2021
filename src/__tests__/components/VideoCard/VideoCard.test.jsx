import React from 'react';
import { BrowserRouter, Link } from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';
import { mount, configure } from 'enzyme';
import { Paper } from '@material-ui/core';

import VideoCard, {
  VideoThumbnail,
  VideoInfo,
  VideoTitle,
  VideoDescription,
} from '../../../components/VideoCard';

describe('VideoCard', () => {
  beforeEach(() => {
    configure({ adapter: new Adapter() });
  });

  it('Renders VideoCard', () => {
    const thumbnail = 'img.jpg';
    const title = 'title';
    const description = 'description';
    const videoId = '123456';
    const wrapper = mount(
      <BrowserRouter>
        <VideoCard
          thumbnail={thumbnail}
          title={title}
          description={description}
          videoId={videoId}
        />
      </BrowserRouter>
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  it('Renders Link', () => {
    const wrapper = mount(
      <BrowserRouter>
        <VideoCard />
      </BrowserRouter>
    );
    expect(wrapper.find(Link)).toBeDefined();
    wrapper.unmount();
  });

  it('Contains Correct Link', () => {
    const id = 'id';
    const otherId = 'otherId';
    const wrapper = mount(
      <BrowserRouter>
        <VideoCard videoId={id} />
      </BrowserRouter>
    );
    expect(wrapper.find(`a[href="/video/${id}"]`).exists()).toBeTruthy();
    expect(wrapper.find(`a[href="/video/${otherId}"]`).exists()).toBeFalsy();
    wrapper.unmount();
  });

  it('Renders Related Video Card Container', () => {
    const wrapper = mount(
      <BrowserRouter>
        <VideoCard />
      </BrowserRouter>
    );
    expect(wrapper.find(Paper)).toBeDefined();
    wrapper.unmount();
  });

  it('Renders VideoThumbnail', () => {
    const wrapper = mount(
      <BrowserRouter>
        <VideoCard />
      </BrowserRouter>
    );
    expect(wrapper.find(VideoThumbnail)).toBeDefined();
    wrapper.unmount();
  });

  it('Contains img', () => {
    const title = 'title';
    const imageUrl = 'img.url';
    const wrapper = mount(
      <BrowserRouter>
        <VideoCard title={title} thumbnail={imageUrl} />
      </BrowserRouter>
    );
    expect(wrapper.find(`img[alt="${title}"]`).exists()).toBeTruthy();
    expect(wrapper.find(`img[src="${imageUrl}"]`).exists()).toBeTruthy();
    wrapper.unmount();
  });

  it('Renders VideoInfo', () => {
    const wrapper = mount(
      <BrowserRouter>
        <VideoCard />
      </BrowserRouter>
    );
    expect(wrapper.find(VideoInfo)).toBeDefined();
    wrapper.unmount();
  });

  it('Renders VideoTitle', () => {
    const wrapper = mount(
      <BrowserRouter>
        <VideoCard />
      </BrowserRouter>
    );
    expect(wrapper.find(VideoTitle)).toBeDefined();
    wrapper.unmount();
  });

  it('Renders VideoTitle', () => {
    const title = 'title';
    const wrapper = mount(
      <BrowserRouter>
        <VideoCard title={title} />
      </BrowserRouter>
    );
    expect(wrapper.find(VideoTitle).text()).toEqual(title);
    wrapper.unmount();
  });

  it('Renders VideoDescription', () => {
    const wrapper = mount(
      <BrowserRouter>
        <VideoCard />
      </BrowserRouter>
    );
    expect(wrapper.find(VideoDescription)).toBeDefined();
    wrapper.unmount();
  });

  it('Renders VideoDescription', () => {
    const description = 'description';
    const wrapper = mount(
      <BrowserRouter>
        <VideoCard description={description} />
      </BrowserRouter>
    );
    expect(wrapper.find(VideoDescription).text()).toEqual(description);
    wrapper.unmount();
  });
});
