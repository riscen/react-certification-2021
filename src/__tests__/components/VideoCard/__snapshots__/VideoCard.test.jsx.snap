// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoCard Renders VideoCard 1`] = `ReactWrapper {}`;
